File: frontend/src/api/trading.ts
import { apiClient } from './client';

export const fetchAccount    = async () => (await apiClient.get('/trading/account')).data;
export const fetchPositions  = async () => (await apiClient.get('/trading/positions')).data;
export const fetchOrders     = async (status = 'all', limit = 50) =>
  (await apiClient.get('/trading/orders', { params: { status, limit } })).data;

export const placeMarketOrder = async (symbol: string, qty: number, side: 'buy' | 'sell') =>
  (await apiClient.post('/trading/orders/market', { symbol, qty, side })).data;

export const placeLimitOrder = async (
  symbol: string, qty: number, side: 'buy' | 'sell', limit_price: number
) => (await apiClient.post('/trading/orders/limit', { symbol, qty, side, limit_price })).data;

export const cancelOrder    = async (id: string) =>
  (await apiClient.delete(`/trading/orders/${id}`)).data;

export const cancelAllOrders = async () =>
  (await apiClient.delete('/trading/orders')).data;

export const closePosition  = async (symbol: string) =>
  (await apiClient.delete(`/trading/positions/${symbol}`)).data;

export const closeAllPositions = async () =>
  (await apiClient.delete('/trading/positions')).data;

File: backend/app/api/v1/trading.py
"""Paper trading routes — all protected by JWT."""
from typing import List
from fastapi import APIRouter, Depends, Query

from app.core.security import get_current_user
from app.services.alpaca_service import alpaca_service
from app.schemas.trading import (
    MarketOrderRequest,
    LimitOrderRequest,
    AccountResponse,
)

router = APIRouter()


@router.get("/account", summary="Get Alpaca paper trading account details")
async def get_account(_user=Depends(get_current_user)):
    return await alpaca_service.get_account()


@router.get("/positions", summary="Get all open positions")
async def get_positions(_user=Depends(get_current_user)):
    return await alpaca_service.get_positions()


@router.get("/orders", summary="Get order history")
async def get_orders(
    status: str = Query("all", description="Filter: all | open | closed"),
    limit: int = Query(50, ge=1, le=200),
    _user=Depends(get_current_user),
):
    return await alpaca_service.get_orders(status=status, limit=limit)


@router.post("/orders/market", summary="Place a market order")
async def place_market_order(body: MarketOrderRequest, _user=Depends(get_current_user)):
    return await alpaca_service.place_market_order(
        symbol=body.symbol.upper(),
        qty=body.qty,
        side=body.side,
    )


@router.post("/orders/limit", summary="Place a limit order")
async def place_limit_order(body: LimitOrderRequest, _user=Depends(get_current_user)):
    return await alpaca_service.place_limit_order(
        symbol=body.symbol.upper(),
        qty=body.qty,
        side=body.side,
        limit_price=body.limit_price,
    )


@router.delete("/orders/{order_id}", summary="Cancel a specific order")
async def cancel_order(order_id: str, _user=Depends(get_current_user)):
    return await alpaca_service.cancel_order(order_id)


@router.delete("/orders", summary="Cancel ALL open orders")
async def cancel_all_orders(_user=Depends(get_current_user)):
    return await alpaca_service.cancel_all_orders()


@router.delete("/positions/{symbol}", summary="Close a specific position")
async def close_position(symbol: str, _user=Depends(get_current_user)):
    return await alpaca_service.close_position(symbol.upper())


@router.delete("/positions", summary="Close ALL open positions")
async def close_all_positions(_user=Depends(get_current_user)):
    return await alpaca_service.close_all_positions()
